paths/paths_unix.go

```diff
@@ -62,10 +63,21 @@ func xdgDataHome() string {
 	return filepath.Join(os.Getenv("HOME"), ".local/share")
 }
 
-func ensureStateDirPerms(dirPath string) error {
-	if filepath.Base(dirPath) != "tailscale" {
+func ensureStateDirPerms(dir string) error {
+	if filepath.Base(dir) != "tailscale" {
 		return nil
 	}
-
-	return os.Chmod(dirPath, 0700)
+	fi, err := os.Stat(dir)
+	if err != nil {
+		return err
+	}
+	if !fi.IsDir() {
+		return fmt.Errorf("expected %v to be a directory; is %v", dir, fi.Mode())
+	}
+	const perm = 0700
+	if fi.Mode().Perm() == perm {
+		// Already correct.
+		return nil
+	}
+	return os.Chmod(dir, perm)
```

Comment:
This would still be a behavior change compared with previous versions: tailscaled will refuse to start, especially problematic if the node is remote.

I understand that merely logging it doesn't do much as nobody looks at the logs, but would like to at least consider the tradeoff of exit vs continue.

Reply:
Until we have a comprehensive integration tests in a read-only builder, read-only disks are not a supported configuration. But this gives people a path forward: they can fix their permissions once manually.

Reply:
On the bug they say that this fix works for them.

Reply:
Okay to submit, @DentonGentry?